feat: prepare data script + pull data into google spreadsheet

lionel-nj opened the pull request:
This PR provides changes to fetch and prepare data for visualization.

# Data

The [rest api provided by Github](https://docs.github.com/en/rest) is heavily leveraged to get data from Github repositories

## What do we get from there?

1. comment counts / quarter-year
2. number of opened issues / quarter-year
3. number of merged PRs / quarter-year
4. open issues count
5. open PR count

This data is gathered for each of the following repositories:
- MobilityData/gtfs-validator
- MobilityData/transit
- MobilityData/gbfs
- NABSA/gbfs
- google/transit

## How is the data transformed into information?
Item from here are counted then classified by quarter of the year.

## Technical
- Calls from Github API are asynchronous
  - [`Octokit`](https://octokit.github.io/rest.js/v18) is the library used to proceed: it has been developed by Github, has great documentation and mechanisms to automatically paginate over HTTP responses.
  - It is necessary to authenticate when executing the implemented scripts in order to avoid HTTP error 403
- aggregated data is exported as `metrics.json` files

## GCP

This PR comes along with the implementation of a pipeline using Google Cloud Platform that enables automation of data crunching.

## View data
- web export: https://docs.google.com/spreadsheets/d/e/2PACX-1vRsOds1l3nNCntabcxqOdvTrVmK1nWjKAsrEoN5gCxB4rqDwrG3j2dDeWgx5pI6m24UGoiG1631fajI/pubhtml
- google spreadsheet: https://docs.google.com/spreadsheets/d/1crXO-U8dUJXiW3TB_SwSJTdiz5hRXAgOGEXyrlxFVjc/edit#gid=303687084

Commit:
define helper methods

Commit:
update gitignore

Commit:
update gitignore

Commit:
update gitignore

Commit:
update gitignore

Commit:
sort dates by ascending order

Commit:
save data locally

Commit:
gitignore

lionel-nj commented:
Per discussion with @carlfredl: it would be nice to have google/transit + MobilityData/transit data aggregated all together.

Commit:
make fetchData an async function

Commit:
use fix versions of dependencies

Commit:
aggregate data per repo and owner

Commit:
make aggregate a new script in package.json

lionel-nj commented:
@carlfredl  updated this PR following your suggestions. 

@maximearmstrong: the PR in question

@heidiguenin according to this PR description, what other indicators could be relevant?

lionel-nj commented:
Please note that nice to have items from https://github.com/MobilityData/Tasks/issues/1170 are not implemented yet.

maximearmstrong reviewed (changes requested):
Overall great job! Just be careful with the style consistency and when possible create constants for the keys/strings that often appear in the code.

Commit:
apply suggestions from code review

- declare symbolic constants
- leverage Standard JS

lionel-nj commented:
Thanks @maximearmstrong for the review! In addition to local installation of Standard JS, what do you think about adding a CI step to check the code syntax? Found [this tool](https://github.com/marketplace/actions/standardjs-lint-checks) that could help us with that.

Commit:
update dependency

Commit:
feat: get count for open PRs and issues

Commit:
try google cloud function

Commit:
other tentative

Commit:
other tentative with gcp

Commit:
fix paths

Commit:
add cloudbuild file to fetch data

Commit:
Merge remote-tracking branch 'origin/prepare-data-script' into prepare-data-script

# Conflicts:
#	src/scripts/FetchRawData.js

Commit:
add require definition

Commit:
modify config file to run task fetch

Commit:
modify config file to run task fetch

Commit:
try including env variable

Commit:
raw data in single json file

Commit:
adjust json field names

Commit:
get results as valid json files

Commit:
add aggregate as step to cloudbuild config file

Commit:
get all metrics in single json file

Commit:
- script to update google sheet

- utils
  - put all constants in the same file

- cloud build file
  - rename cloud build file
  - add step to update google sheet

- metrics
  - gather all comments under one metric

Commit:
separate constants

Commit:
npm install before node .

Commit:
remove google build

Commit:
Revert "remove google build"

This reverts commit e3f25d43

Commit:
use secrets for authentication

Commit:
try other structure for creds

Commit:
fix removal of non-existing folder

Commit:
try with brackets

Commit:
log

Commit:
load env

Commit:
try env export

Commit:
try something else

Commit:
try something else

Commit:
.installed

Commit:
log

Commit:
log

Commit:
use SPREADSHEET_ID

Commit:
remove console outputs

Commit:
apply standard js style

Commit:
add format

Commit:
docs: add README.md

Commit:
use fix versions of dependencies

Commit:
only include year if first quarter while updating spreadsheet

Commit:
update docs and add temporal metrics

Commit:
authenticated request to list users' orgs

Commit:
uncomment code

Commit:
add external comments to google spreadsheet

Commit:
set timeout to 20 mins

Commit:
cache external contributors

Commit:
add dash indicator

Commit:
remove dash indicator

Commit:
put timeout on different steps

Commit:
add "-"  indicator

Commit:
remove "-"

Commit:
timeout next to args

Commit:
timeout to 20 mins

Commit:
timeout to 20 mins

Commit:
timeout to 20 mins

Commit:
debug: do not count prs twice

Commit:
additional documentation

Commit:
additional indicators

Commit:
additional documentation

Commit:
byQuarterYear -> byMonthYear

Commit:
enlarge grid

Commit:
do not update google sheet
